File: app/src/main/java/com/example/david/lists/view/common/ViewModelBase.java
package com.example.david.lists.view.common;

import android.app.Application;

public abstract class ViewModelBase {

    protected final Application application;

    protected ViewModelBase(Application application) {
        this.application = application;
    }


    protected String getStringRes(int resId) {
        return application.getString(resId);
    }

    protected String getStringRes(int resId, Object object) {
        return application.getString(resId, object);
    }
}

File: app/src/main/java/com/example/david/lists/view/common/ViewModelBase.kt
package com.example.david.lists.view.common

import android.app.Application

abstract class ViewModelBase protected constructor(protected val application: Application) {
    protected fun getStringRes(resId: Int): String {
        return application.getString(resId)
    }

    protected fun getStringRes(resId: Int, any: Any): String {
        return application.getString(resId, any)
    }
}
